src/components/Search.js

```diff
@@ -1,159 +1,209 @@
+    const [loadedRecords, setLoadedRecords] = useState([]);
+    const [hasMore, setHasMore] = useState(true);
+    const { username } = useContext(UserContext);
 
-  const navigate = useNavigate();
+    const navigate = useNavigate();
 
-  const getProjects = () => {
-    axiosInstance.get('/projects').then((res) => {
-      const data = res.data/* .slice(69, -1) */;
-      const uniqueTags = uniq(
-        data.flatMap((item) => item.tags.map((tag) => tag.trim()))
-      );
-      const uniqueSkills = uniq(
-        data.flatMap((item) => item.skills.map((skill) => skill.trim()))
-      );
-      setTags(uniqueTags);
-      setSkills(uniqueSkills);
-      setProjects(data);
-    });
-  };
+    const getProjects = useCallback(() => {
+        axiosInstance.get('/projects').then((res) => {
+            const data = res.data; /* .slice(69, -1) */
+            const uniqueTags = uniq(data.flatMap((item) => item.tags.map((tag) => tag.trim())));
+            const uniqueSkills = uniq(data.flatMap((item) => item.skills.map((skill) => skill.trim())));
+            setTags(uniqueTags);
+            setSkills(uniqueSkills);
+            setProjects(data);
+            setLoadedRecords(data.splice(0, 30));
+        });
+    }, []);
 
-  const handleInputChange = ({ target: { id, name, checked } }) => {
-    setQueryParams({
-      ...queryParams,
-      [id]: { ...queryParams[id], [name]: checked },
-    });
-  };
+    const handleInputChange = ({ target: { id, name, checked } }) => {
+        setQueryParams({
+            ...queryParams,
+            [id]: { ...queryParams[id], [name]: checked },
+        });
+    };
 
-  const handleSearch = () => {
-    const { skills, tags } = queryParams;
-    const skillsFilter = Object.keys(skills).filter(
-      (skill) => skills[skill] === true
-    );
-    const tagsFilter = Object.keys(tags).filter((tag) => tags[tag] === true);
+    const handleSearch = () => {
+        const { skills, tags } = queryParams;
+        const skillsFilter = Object.keys(skills).filter((skill) => skills[skill] === true);
+        const tagsFilter = Object.keys(tags).filter((tag) => tags[tag] === true);
 
-    const url = `/projects?skill[]=${encodeURIComponent(
-      skillsFilter
-    )}&tag[]=${encodeURIComponent(tagsFilter)}&search=${encodeURIComponent(
-      searchQuery
-    )}`;
-    axiosInstance.get(url).then((res) => {
-      setProjects(res.data);
-    });
-  };
+        let url = '';
+        if (skillsFilter.length) {
+            skillsFilter.map((skill, index) => {
+                if (index === 0) {
+                    url = `?skill[]=${skill}`;
+                } else url += `&skill[]=${skill}`;
+            });
+        }
+        if (tagsFilter.length) {
+            if (skillsFilter.length) {
+                tagsFilter.map((skill, index) => {
+                    url += `&tag[]=${skill}`;
+                });
+            } else {
+                tagsFilter.map((skill, index) => {
```

Comment:
If there are skills, no more tags? Only tags that are really skills, but no tags that are not skills?